TypeSystem/flow-analysis/reachability_for_in_A03_t03.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion for each statement: If `N` is a for statement of the form
+/// `for (T X in E) S`, `for (var X in E) S`, or `for (X in E) S`, then:
+/// - Let `before(E) = before(N)`
+/// - Let `before(S) = conservativeJoin(after(E), assignedIn(N), capturedIn(N))`
+/// - Let `after(N) = join(before(S), break(S))`
+///
+/// @description Checks that `after(N) = join(before(S), break(S))`. Test that
+/// an assignment in `S` before the `continue` statement is treated by flow
+/// analysis as possibly assigned.
+/// @author sgrekhov22@gmail.com
+
+main() {
+  late int i;
+  if (2 > 1) {
+    for (var j in [42]) {
+      i = 42;
+      continue;
```

Comment:
I think the 'conservative' part of this step in the flow analysis would make the test succeed even in the case where the `continue;` statement occurs _before_ the assignment.

The usual rule could then say "don't test that, because we don't want to depend on the treatment of dead code", but in this particular case it _is_ actually specified that `assignedIn` can't see whether the code is live or dead.

Reply:
There is `reachability_continue_A01_t02.dart` that tests an assignment after `continue` statement. But there were no tests checking an assignment before the `continue`.  Therefore added this one.

Reply:
Thanks!